pkg/storage/utils/decomposedfs/xattrs/xattrs.go

```diff
@@ -178,20 +253,45 @@ func GetInt64(filePath, key string) (int64, error) {
 	return v, nil
 }
 
-// All reads all extended attributes for a node
-func All(filePath string) (map[string]string, error) {
+// All reads all extended attributes for a node, protected by a
+// shared file lock
+func All(filePath string) (attribs map[string]string, err error) {
+	var fileLock *flock.Flock
+
+	fileLock, err = filelocks.AcquireReadLock(filePath)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "xattrs: Unable to lock file for read")
+	}
+	defer func() {
+		rerr := filelocks.ReleaseLock(fileLock)
+
+		// if err is non nil we do not overwrite that
+		if err == nil {
+			err = rerr
+		}
+	}()
+
 	attrNames, err := xattr.List(filePath)
 	if err != nil {
 		return nil, err
 	}
 
-	attribs := make(map[string]string, len(attrNames))
+	var xerrs int = 0
+
+	// error handling: Count if there are errors while reading all attribs.
+	// if there were any, return an error.
+	attribs = make(map[string]string, len(attrNames))
 	for _, name := range attrNames {
-		val, err := xattr.Get(filePath, name)
-		if err != nil {
-			return nil, err
+		if val, xerr := xattr.Get(filePath, name); xerr != nil {
+			xerrs++
+		} else {
+			attribs[name] = string(val)
 		}
-		attribs[name] = string(val)
+	}
+
+	if xerrs > 0 {
+		err = errors.New("Failed to read all xattrs")
 	}
 
 	return attribs, nil
```

Comment:
You shouldn't return nil here. If you do that then the `err` in line 270 will always be nil.